test/unitary/service/pre-post/post.test.ts

```diff
@@ -0,0 +1,165 @@
+    });
+    await post.run();
+    expect(execSpy).toHaveBeenCalledTimes(executedCmds.length);
+    executedCmds.forEach((cmd) => {
+      expect(execSpy).toHaveBeenCalledWith(cmd, process.cwd());
+    });
+  });
+
+  test.each([
+    ["single command", "cmd", ["cmd"]],
+    ["multiple commands", ["cmd1", "cmd2"], ["cmd1", "cmd2"]],
+  ])("With always: %p", async (_title: string, cmds: string | string[], executedCmds: string[]) => {
+    jest.spyOn(BaseConfiguration.prototype, "definitionFile", "get").mockImplementationOnce(() => {
+      return {
+        version: "1.0.1",
+        post: {
+          success: cmds,
+          always: cmds,
+        },
+      };
+    });
+    const execSpy = jest.spyOn(ExecuteCommandService.prototype, "executeCommand").mockImplementationOnce(async (_cmd: string, _cwd?: string) => {
+      return {};
+    });
+    await post.run();
+    expect(execSpy).toHaveBeenCalledTimes(executedCmds.length * 2);
+    [...executedCmds, ...executedCmds].forEach((cmd) => {
+      expect(execSpy).toHaveBeenCalledWith(cmd, process.cwd());
+    });
+  });
+
+  test("no success", async () => {
+    jest.spyOn(BaseConfiguration.prototype, "definitionFile", "get").mockImplementationOnce(() => {
+      return {
+        version: "1.0.1",
+        post: {
+          failure: "cmd",
+        },
+      };
+    });
+    const execSpy = jest.spyOn(ExecuteCommandService.prototype, "executeCommand").mockImplementationOnce(async (_cmd: string, _cwd?: string) => {
+      return {};
+    });
+    await post.run();
+    expect(execSpy).toHaveBeenCalledTimes(0);
+  });
+});
+
+describe("On failure", () => {
+  let post: PostExecutor;
+  
+  beforeAll(() => {
+    Container.reset();
+    Container.set("post.executionSuccess", false);
+  });
+
+  beforeEach(() => {
+    post = Container.get(PostExecutor);
+  });
+
+  test.each([
+    ["single command", "cmd", ["cmd"]],
+    ["multiple commands", ["cmd1", "cmd2"], ["cmd1", "cmd2"]],
+  ])("No always: %p", async (_title: string, cmds: string | string[], executedCmds: string[]) => {
+    jest.spyOn(BaseConfiguration.prototype, "definitionFile", "get").mockImplementationOnce(() => {
+      return {
+        version: "1.0.1",
+        post: {
+          failure: cmds,
+        },
+      };
+    });
+    const execSpy = jest.spyOn(ExecuteCommandService.prototype, "executeCommand").mockImplementationOnce(async (_cmd: string, _cwd?: string) => {
+      return {};
+    });
+    await post.run();
+    expect(execSpy).toHaveBeenCalledTimes(executedCmds.length);
+    executedCmds.forEach((cmd) => {
+      expect(execSpy).toHaveBeenCalledWith(cmd, process.cwd());
+    });
+  });
+
+  test.each([
+    ["single command", "cmd", ["cmd"]],
+    ["multiple commands", ["cmd1", "cmd2"], ["cmd1", "cmd2"]],
+  ])("With always: %p", async (_title: string, cmds: string | string[], executedCmds: string[]) => {
+    jest.spyOn(BaseConfiguration.prototype, "definitionFile", "get").mockImplementationOnce(() => {
```

Comment:
post.ts relies on `configurationservice.getPost` method to get the post commands, so that's the method to mock from here. Consequently, the `configurationservice.getPost` needs to be covered by a new test.